virttest/utils_test/qemu/__init__.py

```diff
@@ -565,59 +576,113 @@ def update_vm_after_unplug(self, vm, dev):
     def hotplug_memory(self, vm, name):
         """
         Hotplug dimm device with memory backend
+        step 1, hotplug memory object
+        step 2, hotplug dimm device
 
         :param vm: VM object
         :param name: memory device name
         """
         devices = vm.devices.memory_define_by_params(self.params, name)
-        for dev in devices:
-            dev_type = "memory"
-            if isinstance(dev, qdevices.Dimm):
-                addr = self.get_mem_addr(vm, dev.get_qid())
-                dev.set_param("addr", addr)
-                dev_type = "pc-dimm"
-            step = "Hotplug %s '%s' to VM" % (dev_type, dev.get_qid())
-            error_context.context(step, logging.info)
-            vm.devices.simple_hotplug(dev, vm.monitor)
-            self.update_vm_after_hotplug(vm, dev)
-        return devices
+        errmsg = "Could not find %s "
+        errmsg += "by memory device name '%s'" % name
+        try:
+            mem = devices[0]
+        except IndexError:
+            raise exceptions.TestError(errmsg % "memory object")
+        try:
+            dimm = devices[1]
+        except IndexError:
+            raise exceptions.TestError(errmsg % "pc-dimm device")
+
+        step = "Hotplug memory object '%s' to VM" % mem.get_qid()
+        error_context.context(step, logging.info)
+        vm.devices.simple_hotplug(mem, vm.monitor)
+        self._update_vm_after_hotplug(vm, mem)
+
+        qid_dimm = dimm.get_qid()
+        dimm.set_param("addr", self.get_mem_addr(vm, qid_dimm))
+        step = "Hotplug pc-dimm device '%s' to VM" % qid_dimm
+        error_context.context(step, logging.info)
+        vm.devices.simple_hotplug(dimm, vm.monitor)
+        self._update_vm_after_hotplug(vm, dimm)
+
+        return self.normalize_mem_size(mem.get_param('size'))
+
+    @error_context.context_aware
+    def hotplug_and_check_memory(self, vm, name):
+        """
+        Hotplug memory device and verify the change
+
+        :param vm: VM object
+        :param name: memory device name
+        """
+        step = "Hotplug memory device '%s'" % name
+        error_context.context(step, logging.info)
+        guest_old_msize = self.get_guest_total_mem(vm)
+        mem_dev_size = self.hotplug_memory(vm, name)
+
+        error_context.context("Verify memory info", logging.info)
+        added_mem_size = self.get_guest_total_mem(vm) - guest_old_msize
```

Comment:
In my mind, we can checked memory immediately in rhel7.0 guest, and we need to online it manually. And we also have case hotplug/unplug memory device when VM is paused status. This change may cause these test failed.

Thanks,
Xu